src/FileIO.chpl

```diff
@@ -308,18 +308,12 @@ module FileIO {
 
     proc lsAnyMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
       var (jsonfile) = payload.splitMsgToTuple(1);
+      var msgArgs = parseMessageArgs(payload, 1);
       
       // Retrieve filename from payload
-      var filename: string;
-      try {
-        filename = jsonToPdArray(jsonfile, 1)[0];
-        if filename.isEmpty() {
-          throw new IllegalArgumentError("filename was empty");  // will be caught by catch block
-        }
-      } catch {
-        var errorMsg = "Could not decode json filenames via tempfile (%i files: %s)".format(
-                                                                                            1, jsonfile);
-        return new MsgTuple(errorMsg, MsgType.ERROR);                                    
+      var filename: string = msgArgs.getValueOf("filename");
+      if filename.isEmpty() {
+        throw new IllegalArgumentError("filename was empty");  // will be caught by catch block
```

Comment:
this is no longer in a catch block, so the comment needs updating. Also should we be throwing an error here or `return new MsgTuple(errorMsg, MsgType.ERROR);`

Reply:
Updated